src/test/java/net/sf/jabref/logic/layout/format/WrapFileLinksTest.java

```diff
@@ -89,4 +91,46 @@ public void testIteratorTwoItems() {
         formatter.setArgument("\\i\n");
         assertEquals("1\n2\n", formatter.format("Test file:test.pdf:PDF;test2.pdf"));
     }
+
+    @Test
+    public void testEndingBracket() {
+        formatter.setArgument("(\\d)");
+        assertEquals("(Test file)", formatter.format("Test file:test.pdf:PDF"));
+    }
+
+    @Test
+    public void testPath() throws IOException {
+        FileLinkPreferences preferences = new FileLinkPreferences(Collections.emptyList(),
+                Collections.singletonList("src/test/resources/pdfs/"));
+        formatter = new WrapFileLinks(preferences);
+        formatter.setArgument("\\p");
+        assertEquals(new File("src/test/resources/pdfs/encrypted.pdf").getCanonicalPath(),
```

Comment:
Can you push the NIO stuff on this branch so we can merge this in even if Oscar is away at the moment?